### ISSUE TITLE
.unitypackage for easy consumption

### ISSUE THREAD

**Issue by maluoi** [microsoft/MixedReality-SpectatorView#268]
Setup is non-trivial for most Unity users I've been working with, it would be fantastically helpful if a .unitypackage could be provided for easy setup.

Git submodules may be great for some users, but honestly, I'm happy when I meet a Unity developer that's familiar with Source Control at all! I've been using git for a few years now, and even I kinda panicked when I saw the setup page.

**Comment by chrisfromwork** [comment]
I think the priority here will be to distribute content via NuGet similar to the MixedReality-Toolkit. @maluoi is the main interest in easing this process for obtaining/referencing dependencies? Are there any concerns with using NuGet compared to .unitypackages?

**Comment by maluoi** [comment]
Unity users are typically comfortable and well versed with .unitypackages, they are the universally standard method for distributing Unity compatible code.

Unity does not support NuGet, nor does it provide quality error messages for those that attempt to use it. The only way to use NuGet with Unity is through a third party extension that isn't available on the Unity Asset Store.

The issue is that the current process for SpectatorView's setup isn't just a little foreign to most Unity developers, it's totally alien! A .unitypackage is by far the easiest way to bundle and distribute Unity code, and by far the easiest way to consume Unity code. The only way it would be easier for users is if that .unitypackage was also hosted on the Unity Asset Store.

For real, it's super easy to make a .unitypackage though. Hands down, 1000x simpler than NuGet.

**Comment by Dustin-Boyle** [comment]
I completely agree with the comments above. Please release unity packages rather than Git submodules, It is the standard for most Unity developers and would save all of us a lot of time. It seems that It would only be one extra short step for the engineer to do so.

**Comment by chrisfromwork** [comment]
I apologize for the delay in response in regards to this work. There has been debate around using unity packages vs MSBuildForUnity amongst Microsoft MixedReality groups. It's looking like we will be moving towards NPM feeds and Unity Packages. This won't quite be the same as distributing a .unitypackage with releases. But it should greatly decrease the pain in consuming Spectator View.

Right now our master branch has some powershell scripts that can be used to generate a Spectator View unity package + dependencies. We have not tested them widely outside our team but it may be able to unblock folks for now. We are going to look into setting up a NPM feed for spectator view, but it will require having some of our dependencies (Azure Spatial Anchors, MRTK, etc.) also distribute Unity Packages. This will take more time.

**Comment by pianita** [comment]
Hello!
Are there any news regarding this issue?
I tried creating a unity package using the instructions on this repository but get a lot of errors, e.g.:
- not finding msbuild.exe under the predefined path because i am using visual studio community not enterprise
- not finding nuget

A simple unitypackage would be great... :)
